tests/common_tests/chowdsp_core_test/EndOfScopeActionTest.cpp

```diff
@@ -3,11 +3,26 @@
 
 TEST_CASE ("End Of Scope Action Test", "[common][functional]")
 {
-    int x = 4;
+    SECTION ("Simple usage")
     {
-        const auto _ = chowdsp::runAtEndOfScope ([&x]
-                                                 { x = 0; });
-        REQUIRE (x == 4);
+        int x = 4;
+        {
+            const auto _ = chowdsp::runAtEndOfScope ([&x]
+                                                     { x = 0; });
+            REQUIRE (x == 4);
+        }
+        REQUIRE (x == 0);
+    }
+
+    SECTION ("Moved-from object doesn't invoke callback")
+    {
+        int lambdaCalled { 0 };
+        {
+            chowdsp::EndOfScopeAction endOfScopeAction { [&lambdaCalled]
+                                                         { ++lambdaCalled; } };
+            auto movedTo = std::move (endOfScopeAction);
+            REQUIRE (lambdaCalled == 0);
+        }
+        REQUIRE (lambdaCalled == 1);
```

Comment:
It seems like this test only tests the "move constructor" code path, and not the "move assignment" code path. Ideally, the move assignment path would be tested as well (including both branches of the if statement in that path).

That said, I remember trying to write a test to explicitly test move assignment for something a while ago, and finding it surprisingly difficult... so no worries if that turns out to be the case here.

Reply:
I've added a test now that should the move assignment as well!